File: server/php/src/Application/Domain/Payments/PaymentDetails/SurchargeSpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class SurchargeSpecificOutput
{
    public ?string $mode = null;
    public ?AmountOfMoney $surchargeAmount = null;
    public ?SurchargeRate $surchargeRate = null;

    public function toArray(): array
    {
        return [
            'mode' => $this->mode,
            'surchargeAmount' => ($this->surchargeAmount ?? new AmountOfMoney())->toArray(),
            'surchargeRate' => ($this->surchargeRate ?? new SurchargeRate())->toArray(),
        ];
    }
}

File: server/python/app/application/domain/payments/payment_details/surcharge_specific_output.py
from dataclasses import dataclass
from typing import Optional

from app.application.domain.payments.payment_details.amount_of_money import AmountOfMoney
from app.application.domain.payments.payment_details.surcharge_rate import SurchargeRate

@dataclass
class SurchargeSpecificOutput:
    mode: Optional[str] = None
    surcharge_amount: Optional[AmountOfMoney] = None
    surcharge_rate: Optional[SurchargeRate] = None